diff --git a/node_modules/modern-react-qr-reader/lib/index.js b/node_modules/modern-react-qr-reader/lib/index.js
index 27330bc..f02698a 100644
--- a/node_modules/modern-react-qr-reader/lib/index.js
+++ b/node_modules/modern-react-qr-reader/lib/index.js
@@ -235,7 +235,11 @@ module.exports = (_temp = _class = function (_Component) {
 
       var streamTrack = stream.getTracks()[0];
       // Assign `stopCamera` so the track can be stopped once component is cleared
-      this.stopCamera = streamTrack.stop.bind(streamTrack);
+      this.stopCamera = () => {
+        setTimeout(() => {
+          streamTrack.stop();
+        }, 2000);
+      }
 
       preview.addEventListener('loadstart', this.handleLoadStart);
 
diff --git a/node_modules/modern-react-qr-reader/src/index.js b/node_modules/modern-react-qr-reader/src/index.js
index 879993e..3e5f13c 100644
--- a/node_modules/modern-react-qr-reader/src/index.js
+++ b/node_modules/modern-react-qr-reader/src/index.js
@@ -195,7 +195,11 @@ module.exports = class Reader extends Component {
 
     const streamTrack = stream.getTracks()[0]
     // Assign `stopCamera` so the track can be stopped once component is cleared
-    this.stopCamera = streamTrack.stop.bind(streamTrack)
+    this.stopCamera = () => {
+      setTimeout(() => {
+        streamTrack.stop();
+      }, 2000);
+    }
 
     preview.addEventListener('loadstart', this.handleLoadStart)
 
